charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info.py

```diff
@@ -217,24 +228,26 @@ def __init__(self, provider_charm, relation_name: str = DEFAULT_RELATION_NAME):
         self.provider_charm = provider_charm
         self.relation_name = relation_name
 
-    def send_gateway_relation_data(self, gateway_name: str, gateway_namespace: str) -> None:
-        """Updates the relation data bag with data from the localGateway.
+    def send_gateway_relation_data(self, gateway_name: str, gateway_namespace: str, gateway_up: bool = True) -> None:
+        """Updates the relation data bag of any related applications with data from the localGateway.
+
+        This method will complete successfully even if there are no related applications.
+
         Args:
             gateway_name (str): the name of the Gateway the provider knows about
             gateway_namespace(str): the namespace of the Gateway the provider knows about
+            gateway_up (bool): (optional) the status of the Gateway.  Assumed to be True by the
```

Comment:
```suggestion
            gateway_up (bool): (optional) the status of the Gateway. Defaults to True if not provided.
```

Reply:
Added a note to the definition in the lib docstring.  Does that work better?